src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -799,6 +804,765 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetExternalIPv6
+            Encoding::LittleEndian::Put64(*buffer, activeDataset.mActiveTimestamp);
+        }
+    }
+    break;
+
+    case ZCL_PENDING_TIMESTAMP_ATTRIBUTE_ID: {
+        type = kTLVType_UnsignedInteger;
+        if (otDatasetIsCommissioned(mOTInst))
+        {
+            ReadLength = sizeof(uint64_t);
+            *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+            VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+            otOperationalDataset activeDataset;
+            otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
+            VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
+            Encoding::LittleEndian::Put64(*buffer, activeDataset.mPendingTimestamp);
+        }
+    }
+    break;
+
+    case ZCL_DELAY_ATTRIBUTE_ID: {
+        type = kTLVType_UnsignedInteger;
+        if (otDatasetIsCommissioned(mOTInst))
+        {
+            ReadLength = sizeof(uint32_t);
+            *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+            VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+            otOperationalDataset activeDataset;
+            otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
+            VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
+            Encoding::LittleEndian::Put32(*buffer, activeDataset.mDelay);
+        }
+    }
+    break;
+
+    case ZCL_SECURITY_POLICY_ATTRIBUTE_ID: {
+        type = kTLVType_Structure;
+        if (otDatasetIsCommissioned(mOTInst))
+        {
+            ReadLength = sizeof(otSecurityPolicy);
+            *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+            VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+            otOperationalDataset activeDataset;
+            otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
+            VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
+            memcpy(*buffer, &activeDataset.mSecurityPolicy, sizeof(otSecurityPolicy));
+        }
+    }
+    break;
+
+    case ZCL_DIAG_CHANNEL_MASK_ATTRIBUTE_ID: {
+        type = kTLVType_ByteString;
+        if (otDatasetIsCommissioned(mOTInst))
+        {
+            ReadLength = (sizeof(uint32_t) * 2) + 1; // For all Hex characters + NULL
+            *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+            VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+            otOperationalDataset activeDataset;
+            otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
+            VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
+
+            // In the resultant Octet string, the most significant bit of the left-most byte indicates channel 0
+            uint32_t swapedChannelMask = Encoding::Swap32(activeDataset.mChannelMask);
```

Comment:
What is the endianness of `activeDataset.mChannelMask` and how are its bits arranged?

Reply:
mChannelMask is a a uint32 bitfield, channel 0 is bit0 up to channel 25 bit 25

The spec specify this for the bit order `The channels are represented in most significant bit order, with bit value 1 meaning selected, bit value 0 meaning unselected. For example, the most significant bit of the left-most byte indicates channel 0. If channel 0 and channel 10 are selected, the mask would be: 80 20 00 00.`

Reply:
OK, thank you.  So concretely, for channel N the bit value in mChannelMask is gotten by doing `mChannelMask & (1 << N)`, right?

Let's say only channel 0 is selected.  Then the value of `mChannelMask` is `0x00000001`.  After calling Swap32, the value will be `0x01000000`.  The value that needs to end up getting encoded is the array of bytes with values `{ 0x80, 00, 00, 00 }`, right?

In particular, what's needed here is swapping the order of the _bits_, not just of the bytes....